apps/web/src/components/ZapLiquidityWidget.tsx

```diff
@@ -58,16 +70,36 @@ export const ZapLiquidityWidget: React.FC<ZapLiquidityProps> = ({
 
   const { toastSuccess } = useToast()
 
+  const poolAddress = useMemo(() => pool && Pool.getAddress(pool.token0, pool.token1, pool.fee), [pool])
+
   const [isModalOpen, setIsModalOpen] = useState(false)
 
-  const poolAddress = useMemo(() => pool && Pool.getAddress(pool.token0, pool.token1, pool.fee), [pool])
+  const [isWalletModalOpen, setIsWalletModalOpen] = useState(false)
 
-  const handleOnDismiss = useCallback(() => {
-    setIsModalOpen(false)
-  }, [])
+  const [depositTokens, setDepositTokens] = useState<string>('')
+
+  const [amounts, setAmounts] = useState<string>('')
+
+  const handleWalletModalOnDismiss = useCallback(() => setIsWalletModalOpen(false), [])
+
+  const handleOnWalletConnect = useCallback(() => setIsWalletModalOpen(true), [])
 
   const handleOnClick = useCallback(() => {
+    setDepositTokens(
+      initDepositToken === InitDepositToken.BASE_CURRENCY
+        ? baseCurrency?.isNative
```

Comment:
const getTokenAddress = (token?: { isNative?: boolean; wrapped?: { address?: string } }) => token?.isNative ? NATIVE_CURRENCY_ADDRESS : token?.wrapped?.address || '';

Do we have a function similar `getTokenAddress` in SmartRouter? Seems lots of address testing .